contracts/IbbtcVaultZap.sol

```diff
@@ -146,8 +146,62 @@ contract IbbtcVaultZap is PausableUpgradeable {
             );
     }
 
+    function _vaultShares(uint256 _amount)
+        internal
+        view
+        returns (uint256 shares)
+    {
+        uint256 totalSupply = IBBTC_VAULT.totalSupply();
+        shares = _amount;
+        if (totalSupply != 0) {
+            shares = (_amount.mul(totalSupply)).div(IBBTC_VAULT.balance());
+        }
+    }
+
+    function _calcIbbtcMint(uint256[2] memory _amounts)
+        internal
+        view
+        returns (uint256 bBTC)
+    {
+        uint256 fee;
+        uint256 lp = CURVE_REN_POOL.calc_token_amount(_amounts, true);
+        uint256 sett = lp.mul(1e18).div(RENCRV_VAULT.getPricePerFullShare());
+        (bBTC, fee) = SETT_PEAK.calcMint(0, sett);
+    }
+
     /// ===== Public Functions =====
 
+    function calcMint(uint256[4] calldata _amounts, bool _mintIbbtc)
+        public
+        view
+        returns (uint256)
+    {
+        uint256[4] memory depositAmounts;
+
+        for (uint256 i = 0; i < 4; i++) {
+            if (_amounts[i] > 0) {
+                if (!_mintIbbtc || i == 0 || i == 3) {
+                    depositAmounts[i] = _amounts[i];
+                }
+            }
+        }
+        if (_mintIbbtc && (_amounts[1] > 0 || _amounts[2] > 0)) {
+            // Use renbtc and wbtc to mint ibbtc
+            // NOTE: Can change to external zap if implemented
+            depositAmounts[0] = depositAmounts[0].add(
+                _calcIbbtcMint([_amounts[1], _amounts[2]])
+            );
+        }
```

Comment:
We can probably wrap this logic in an internal function since it's duplicated

Reply:
one place there is `_calcIbbtcMint` used and other `_renZapToIbbtc`

Reply:
Makes sense